app/forms/supplementary/ItemType.scala

```diff
@@ -53,74 +63,58 @@ case class ItemType(
+    (JsPath \ "combinedNomenclatureCode").read[String] and
+      (JsPath \ "taricAdditionalCode").read[Seq[String]] and
+      (JsPath \ "nationalAdditionalCode").read[Seq[String]] and
+      (JsPath \ "descriptionOfGoods").read[String] and
+      (JsPath \ "cusCode").readNullable[String] and
+      (JsPath \ "statisticalValue").read[String]
+  )(ItemType.apply _)
+
+  implicit val writes: Writes[ItemType] = (
+    (JsPath \ "combinedNomenclatureCode").write[String] and
+      (JsPath \ "taricAdditionalCode").write[Seq[String]] and
+      (JsPath \ "nationalAdditionalCode").write[Seq[String]] and
+      (JsPath \ "descriptionOfGoods").write[String] and
+      (JsPath \ "cusCode").writeNullable[String] and
+      (JsPath \ "statisticalValue").write[String]
+  )(unlift(ItemType.unapply))
 
-  private val combinedNomenclatureCodeMaxLength = 8
-  private val taricAdditionalCodeLength = 4
-  private val nationalAdditionalCodeMaxLength = 4
-  private val descriptionOfGoodsMaxLength = 280
-  private val cusCodeLength = 8
-  private val statisticalValueMaxLength = 15
-  private val statisticalValueDecimalPlaces = 2
+  val taricAdditionalCodesKey = "taricAdditionalCode"
+  val nationalAdditionalCodesKey = "nationalAdditionalCode"
 
   val mapping = Forms.mapping(
-    "combinedNomenclatureCode" -> text()
-      .verifying("supplementary.itemType.combinedNomenclatureCode.error.empty", nonEmpty)
-      .verifying(
-        "supplementary.itemType.combinedNomenclatureCode.error.length",
-        isEmpty or noLongerThan(combinedNomenclatureCodeMaxLength)
-      )
-      .verifying("supplementary.itemType.combinedNomenclatureCode.error.specialCharacters", isEmpty or isAlphanumeric),
-    "taricAdditionalCode" -> optional(
-      text()
-        .verifying(
-          "supplementary.itemType.taricAdditionalCodes.error.length",
-          hasSpecificLength(taricAdditionalCodeLength)
-        )
-        .verifying("supplementary.itemType.taricAdditionalCodes.error.specialCharacters", isAlphanumeric)
-    ),
-    "nationalAdditionalCode" -> optional(
-      text()
-        .verifying(
-          "supplementary.itemType.nationalAdditionalCode.error.length",
-          noLongerThan(nationalAdditionalCodeMaxLength)
-        )
-        .verifying("supplementary.itemType.nationalAdditionalCode.error.specialCharacters", isAlphanumeric)
-    ),
-    "descriptionOfGoods" -> text()
-      .verifying("supplementary.itemType.description.error.empty", nonEmpty)
-      .verifying(
-        "supplementary.itemType.description.error.length",
-        isEmpty or noLongerThan(descriptionOfGoodsMaxLength)
-      ),
-    "cusCode" -> optional(
-      text()
-        .verifying("supplementary.itemType.cusCode.error.length", hasSpecificLength(cusCodeLength))
-        .verifying("supplementary.itemType.cusCode.error.specialCharacters", isAlphanumeric)
-    ),
+    "combinedNomenclatureCode" -> text(),
+    taricAdditionalCodesKey -> default(seq(text()), Seq.empty),
```

Comment:
Why this? `default(seq(text()), Seq.empty)`

Reply:
Because without default() we would bind empty string into the sequence.